src/core/config/logging.h

```diff
@@ -277,6 +277,16 @@
 #define OPENTHREAD_CONFIG_LOG_PREPEND_REGION 1
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_LOG_COLOR_ENABLE
+ *
+ * Define as 1 to enable color in logs.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_LOG_COLOR_ENABLE
+#define OPENTHREAD_CONFIG_LOG_COLOR_ENABLE 0
+#endif
+
```

Comment:
While I like  the idea of adding color in posix logging, I am not sure about the configuration through the `config/logging.h` and common-switches makefile.
- In general, definitions under `core/config/` should configure the OT core behavior itself (and be independent of and applicable to all/multiple platforms).
- Other definitions in the `config/logging.h` do follow this (they impact how the logging is implemented with OT core). 
- However `OPENTHREAD_CONFIG_LOG_COLOR_ENABLE` does not change anything within core, and it needs to be handled by the platform code (it's used by `otPlatLog()` from `posix` example which is part of platform code).
- Also this config is only applicable to posix platform example.

My suggestion is to add this in `platform-config` specific to posix example (`example/platforms/posix/platform-config.h` file).

Reply:
Agree with @abtink's point above.

Some other points worth mentioning:

- rsyslog escapes control characters by default. There is a
 global `/etc/rsyslog.conf` configuration [EscapeControlCharactersOnReceive](https://www.rsyslog.com/doc/v8-stable/configuration/input_directives/rsconf1_escapecontrolcharactersonreceive.html), but that will affect all input received by rsyslog.
  ```
  Sep 20 21:54:26 558cfb0e1e95 ot-cli-ftd[11946]: #033[1;36m[1] [NOTE]-MLE-----: Role Disabled -> Detached#033[0m
  ```

- Why not leverage existing colorization tools such as [ccze](https://github.com/cornet/ccze)? Note that this does not require any change to OpenThread and rsyslog configuration. Example screenshot:
  <img width="693" alt="Screen Shot 2019-09-20 at 3 11 03 PM" src="https://user-images.githubusercontent.com/16142208/65361565-e4c39880-dbb8-11e9-85ab-0fa664af9b5e.png">